### ISSUE TITLE
Is there any API to get allowed value of permission?

### ISSUE THREAD

**Issue by Hajime-san** [denoland/deno#22985]
I'm using the nice library [dax](https://github.com/dsherret/dax) heavily to write shell-like script.
BTW, it requires all environment variable [inside](https://github.com/dsherret/dax/blob/ed9fbd1f0b7aef914f78efc384dba55b719880b9/src/command.ts#L1076) while it needs only `$PATH` environment variable to run `ls`.

as follows

```ts
import $ from 'dax';

await $`ls`;
```

```bash
% deno run --allow-env=PATH --allow-read --allow-run=ls ls.ts
┌ ⚠️  Deno requests env access.
├ Run again with --allow-env to bypass this prompt.
└ Allow? [y/n/A] (y = yes, allow; n = no, deny; A = allow all env permissions) > 
```

So, I wish if an API to get allowed value of permission.

I'm not clear whether it passes security requirements model of Deno.

Thanks!

**Comment by dsherret** [comment]
@Hajime-san can you rephrase your question? There is a `Deno.permissions.query` API, but I'm not sure that's what you're looking for: https://docs.deno.com/runtime/manual/runtime/permission_apis/

**Comment by Hajime-san** [comment]
Thank you for replying!

You suggest the `Deno.permissions.query` API is a way of to get value that allowed by user.
However, it requires that library or application author to write codes with predicting what permission value will be allowed in detail.
For instance, the [Deno.Env.toObject](https://docs.deno.com/api/deno/~/Deno.Env.toObject) API returns all environment value with giving `--allow-env` permission.
So, I'm trying to say that  like `Deno.env.onAllowed` API extends Event interface (or WICG Proposal [Observable](https://wicg.github.io/observable/)) it will return permission allowed environment value at that point.
